### PR TITLE
Add support for non btrfs devices in GetDirFsDevice

### PR DESCRIPTION
**Pull request by saschagrunert**
We can create a `*DeviceInfo` from the mount even if the `FSType` is not
`btrfs`.

Refers to: https://github.com/kubernetes/kubernetes/issues/94825
Testing in: https://github.com/kubernetes/kubernetes/pull/100451

### CODE REVIEW

**Comment by saschagrunert** [comment]
CI in kubernetes/kubernetes#100451 was green, removing the WIP from here.

File: fs/fs.go [hunk 1]
```diff
@@ -582,14 +582,19 @@ func (i *RealFsInfo) GetDirFsDevice(dir string) (*DeviceInfo, error) {
...
-			klog.Warningf("%s", err)
-		} else {
-			return &DeviceInfo{mnt.Source, uint(major), uint(minor)}, nil
+	if found && strings.HasPrefix(mnt.Source, "/dev/") {
+		major, minor := mnt.Major, mnt.Minor
+
+		if mnt.FSType == "btrfs" && major == 0 {
+			major, minor, err = getBtrfsMajorMinorIds(mnt)
+			if err != nil {
+				klog.Warningf("%s", err)
```
Comment (iwankgb) [inline fs/fs.go:591]: Could you add some context to this error message, please?

**Review by iwankgb — COMMENTED** [review]
LGTM but I would appreciate if you improved log message a bit.

File: fs/fs.go (hunk 1 above)
Comment (saschagrunert) [reply fs/fs.go:591]: Done :)

**Comment by saschagrunert** [comment]
> LGTM but I would appreciate if you improved log message a bit.

Done, thank you for the hint! :pray:

**Review by iwankgb — APPROVED** [review]

**Pull request merged**